prow/private_deck.md

```diff
@@ -0,0 +1,188 @@
+# Setting up Private Deck
+
+## 1) Set up TenantIDs for ProwJobs and Repos
+
+Once you set up a tenantID, all prowjobs labled with that tenantID will only be visable on Deck instances created with the same tenantID. If you already have prowjobs that you don't want to lose access to on Deck, do this step last. If not, do it first to make sure prowjobs you want to keep sequestered do not appear on other instances of Deck.
+
+The recommended way to add TenantIDs to Prowjobs based on org/repo or cluster is through prowjob_default_entries in the prow config. This will apply the tenant ID to jobs with matching cluster AND repo. If you want to do cluster OR repo, create two entries in the config and use "*" for either field.
+```yaml
+prowjob_default_entries:
+  - cluster: "build-private"
+    repo: "*"
+    config:
+      tenant_id: 'private'
+  - cluster: '*'
+    repo: 'private'
+    config:
+      tenant_id: 'private'
+
+```
+This configuration is used both to apply tenantIDs to prowjobs, but is also used by Deck to filter out Tide information from orgRepos with tenantIDs that do not match. So even if you can lable all your prowjobs using cluster, make sure that all of your repos are given a tenantID as well.
+
+### Override TenantIDs
+
+You can also define a tenantID for a given Prowjob by defining it in the Prowjob spec under spec.ProwJobDefault. This will override the tenantID assigned via prowjob defaults.
+
+## 2) Create a new Service Account and Bind it
+
+```
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  namespace: default
+  name: <SA_NAME>
+  annotations:
+    "iam.gke.io/gcp-service-account": "..."
+```
+
+``` bash
+gcloud iam service-accounts add-iam-policy-binding \
+  --project=PROJECT \
+  --role=roles/iam.workloadIdentityUser \
+  --member=serviceAccount:K8S_PROJECT.svc.id.goog[SOMEWHERE/SOMETHING] \
+  SOMEBODY@PROJECT.iam.gserviceaccount.com
+
+```
+## 3) (Optional) Create a new OauthApp for authentication
+
+If you want to make the new Deck instance Private, create a new oauth app.
+
+### Creating oauth secrets
+
+You will need to create two secrets for the oauth app. The easiest way to do this is to create a secret in Secret Manager and create two External Secrets that are bound to it.
+
+The Secret in GCS will need to be in JSON and contain the following fields. Populate the secret with information from the newly created oauth app:
+
+```
+{"client_id":"...",
+"client_secret":"...",
+"redirect_url":"https://DOMAIN/oauth2/callback",
```

Comment:
We will need to know the DOMAIN ahead of time, should we mention that in previous step?